src/Compilers/CSharp/Portable/FlowAnalysis/DefiniteAssignment.cs

```diff
@@ -81,6 +81,25 @@ internal partial class DefiniteAssignmentPass : LocalDataFlowPass<
         /// </summary>
         private readonly PooledHashSet<Symbol> _writtenVariables = PooledHashSet<Symbol>.GetInstance();
 
+        /// <summary>
+        /// Struct fields that are implicitly initialized, due to being used before being written, or not being written at an exit point.
+        /// </summary>
+        private PooledHashSet<FieldSymbol>? _implicitlyInitializedFieldsOpt;
+
+        private void AddImplicitlyInitializedField(FieldSymbol field)
+        {
+            Debug.Assert(TrackImplicitlyInitializedFields);
+            (_implicitlyInitializedFieldsOpt ??= PooledHashSet<FieldSymbol>.GetInstance()).Add(field);
+        }
+
+        private bool TrackImplicitlyInitializedFields
+        {
+            get
+            {
+                return _requireOutParamsAssigned && !this._emptyStructTypeCache._dev12CompilerCompatibility;
```

Comment:
>_requireOutParamsAssigned [](http://example.com/codeflow?start=23&length=25)

Sorry, we talked about ​`_requireOutParamsAssigned​` last week, but I still have a question. Shouldn't we check whether we're analyzing a constructor instead?

Reply:
Have clarified offline. There are scenarios where '_requireOutParamsAssigned' is false, yet 'CurrentSymbol' is a struct constructor.

Reply:
Could we add the struct constructor check? Alternatively, we could add a "struct constructor" assertion to `AddImplicitlyInitializedField`.